src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -514,5 +514,77 @@ public static class Diagnostics {
 			isEnabledByDefault: true,
 			description: "We require that event types be sealed."
 		);
+
+		public static readonly DiagnosticDescriptor MutableBaseType = new DiagnosticDescriptor(
+			id: "D2L0061",
+			title: "Base class for immutable type must be [Immutable] (or [ImmutableBaseClass])",
+			messageFormat: "{0}'s base class is {1} which is missing [Immutable] (or, more weakly, [ImmutableBaseClass]).",
+			category: "Immutability",
+			defaultSeverity: DiagnosticSeverity.Error,
+			isEnabledByDefault: true
+		);
+
+		public static readonly DiagnosticDescriptor ArraysAreMutable = new DiagnosticDescriptor(
```

Comment:
Does this need a special diagnostic? Or are you calling it out separately because you can't `[Immutable]` so the language in `D2L0066` doesn't apply?

Does it need to be separate from `D2L0063` though? "This is a mutable thing that can't be made immutable, choose something else"

Reply:
> Does this need a special diagnostic? Or are you calling it out separately because you can't [Immutable] so the language in D2L0066 doesn't apply?

It's a separate branch of code, but the reason you gave is good.

> Does it need to be separate from D2L0063 though? "This is a mutable thing that can't be made immutable, choose something else"

D2L0063 is about like `private readonly dynamic m_magic`.

Reply:
(The error messages don't _have_ to be this granular (they're very much not right now) but the steps to fix are very different so I think it's worth it.)

Reply:
I guess since you're suggesting `ImmutableArray`, k